clang/test/dpct/syncthreads.cu

```diff
@@ -149,4 +161,129 @@ __global__ void test9(S4 a) {
   S4 b(a);
   // CHECK:item_ct1.barrier();
   __syncthreads();
-}
\ No newline at end of file
+}
+
+extern __shared__ float extern_local_decl[];
+
+__global__ void test10(float *arg1_ptr, int arg2_scalar, int arg3_scalar) {
+  int var1 = blockIdx.x * blockDim.x + threadIdx.x;
+  if (var1 < arg3_scalar) {
```

Comment:
also, we cannot determine syncthread call one by one.
We need to analysis in an iteration, because previous change (global => local) will affect currect analysis.